### PR TITLE
feat(fork-network): override configs at SetValidatorsCmd

### PR DESCRIPTION
**Pull request by Longarithm**
I've made two observations:
* `num_seats` is not propagated to forknet in `set-validators` command, like it was done in `amend-genesis`
* the actual node dir preparation happens at `set-validators`, not `init_configs`

So I suggest to move epoch config overrides to `set-validators` as well. It looks very natural. I think we can always assume that `genesis_protocol_version` is the first version, and then pass epoch config overrides through `new-test` to `set-validators`. I used this for forknet resharding and it worked well.

If epoch config overrides happen at init_configs, then, after init and before new-test, forknet nodes will be in weird state where epoch configs are already overridden but genesis isn't.

### CODE REVIEW

File: tools/fork-network/src/cli.rs [hunk 1]
```diff
@@ -527,6 +532,52 @@ impl ForkNetworkCommand {
...
+        let epoch_config_dir = home_dir.join("epoch_configs");
+        if epoch_config_dir.exists() {
+            std::fs::remove_dir_all(epoch_config_dir.clone())?;
+        }
+        std::fs::create_dir_all(epoch_config_dir.clone()).with_context(|| {
+            anyhow::anyhow!("Failed to create directory {:?}", epoch_config_dir)
+        })?;
+
+        let base_epoch_config_store =
+            EpochConfigStore::for_chain_id(near_primitives::chains::MAINNET, None)
```
Comment (marcelo-gonzalez) [inline tools/fork-network/src/cli.rs:552]: Could be good to be aware somehow of what chain we forked to create the state in the first place (could have been testnet, or some other random net). Maybe could just derive this epoch config from the starting genesis? Anyway it's not a big deal for now, so we can just worry about this later. But maybe add a TODO?

File: tools/fork-network/src/cli.rs [hunk 2]
```diff
@@ -527,6 +532,52 @@ impl ForkNetworkCommand {
...
+                config.validator_selection_config.num_chunk_producer_seats = *num_seats;
+                config.validator_selection_config.num_chunk_validator_seats = *num_seats;
+            }
+            new_epoch_configs.insert(version, config);
+        }
+        let first_config = new_epoch_configs.get(&first_version).unwrap().clone();
+
+        let new_arc_epoch_configs = new_epoch_configs
+            .into_iter()
+            .map(|(version, config)| (version, Arc::new(config)))
```
Comment (marcelo-gonzalez) [inline tools/fork-network/src/cli.rs:568]: why not just make them Arcs in the first loop?

**Review by marcelo-gonzalez — APPROVED** [review]

File: tools/fork-network/src/cli.rs (hunk 1 above)
Comment (Longarithm) [reply tools/fork-network/src/cli.rs:552]: 1) Do we ever need to fork any chain other than mainnet?
2) Creating epoch config from genesis is actually what we want to get rid of: #11265.

Comment (marcelo-gonzalez) [reply tools/fork-network/src/cli.rs:552]: 1. eh I guess it's not impossible. maybe one day something interesting happens on testnet and we want to try it out in forknet
2. +1

**Pull request merged**